nxontology/ontology.py

```diff
@@ -30,9 +30,17 @@ class NXOntology(Freezable, Generic[Node]):
 
     def __init__(self, graph: nx.DiGraph | None = None):
         self.graph = nx.DiGraph(graph)
+        if graph is None:
+            # Store the nxontology version that created the graph as metadata,
+            # in case there are compatability issues in the future.
+            self._add_nxontology_metadata()
         self.check_is_dag()
         self._node_info_cache: dict[Node, Node_Info[Node]] = {}
 
+    def _add_nxontology_metadata(self) -> None:
+        self.graph.graph["nxontology_version"] = get_nxontology_version()
+        self.graph.graph["nxontology_created"] = get_datetime_now()
```

Comment:
One downside is that the checksum of an nxontology output file changes every time its created, even when the underlying data hasn't changed. This is usually fine, but won't be ideal for the compressed outputs in https://github.com/related-sciences/nxontology-data.